lib/src/license_detection/primary_filter.dart

```diff
@@ -0,0 +1,37 @@
+import 'package:pana/src/license_detection/license.dart';
+
+/// Filters the corpus licenses and
+/// returns a list of [License] which might
+/// be a possible match.
+List<License> filter(Map<String, int> occurences, List<License> knownLicenses) {
+  var possibleLicenses = <License>[];
+
+  knownLicenses.forEach((license) {
+    if (tokenSimilarity(occurences, license.occurences) >= 0.5) {
+      possibleLicenses.add(license);
+    }
+  });
```

Comment:
```suggestion
  knownLicenses.where((license) => tokenSimilarity(occurences, license.occurences) >= 0.5);
```

You can probably even do:
```suggestion
  return List.unmodifiable(knownLicenses.where((license) {
    return tokenSimilarity(occurences, license.occurences) >= 0.5);
  });
```

Using `List.where` returns an iterable, so it's often necessary to call `List.where(...).toList()`, but here you prefer an unmodifiable list, so just using `List.unmodifiable` is fine (it also accepts an iterable).